server/server_admin.go

```diff
@@ -70,6 +70,44 @@ func (s *Server) handleUsersAdd(w http.ResponseWriter, r *http.Request, v *visit
 	}
 	return s.writeJSON(w, newSuccessResponse())
 }
+func (s *Server) handleUsersUpdate(w http.ResponseWriter, r *http.Request, v *visitor) error {
+	req, err := readJSONWithLimit[apiUserAddOrUpdateRequest](r.Body, jsonBodyBytesLimit, false)
+	if err != nil {
+		return err
+	} else if !user.AllowedUsername(req.Username) || req.Password == "" {
```

Comment:
This implies that this API is only for changing the password. What if we only want to change the tier?

 `{"username": "ben", "tier": "pro"}`

I know that the API isn't originally for that, and it's undocumented, but why not :-)